### PR TITLE
rumqttc: convert all direct index access to `get_mut`

### LINKED ISSUE #201: Replace index access with get in state
Replace indexing in broker and client [state](https://github.com/bytebeamio/rumqtt/blob/master/rumqttc/src/state.rs#L221) to prevent crashes due to bad brokers or clients.

### PR DESCRIPTION
**Pull request by 123vivekr**
Fixes #201

### CODE REVIEW

**Review by de-sh — COMMENTED** [review]
The way pkids work is they are between `0 <= pkid < max_inflight`. Thus a check that looks like `pkid > max_inflight` could easily tell us if received pkid is going to panic or not.

File: rumqttc/src/state.rs [hunk 1]
```diff
@@ -218,9 +218,13 @@ impl MqttState {
...
-        let v = match mem::replace(&mut self.outgoing_pub[puback.pkid as usize], None) {
-            Some(_) => {
-                self.inflight -= 1;
+        let v = match self.outgoing_pub.get_mut(puback.pkid as usize) {
+            Some(pkid) => {
+                if let Some(_) = pkid {
+                    self.inflight -= 1;
+                    *pkid = None;
+                }
+
```
[the remark this answers is not in the document]
Comment (123vivekr) [reply rumqttc/src/state.rs:227]: Yes, it's better now

File: rumqttc/src/state.rs [hunk 2]
```diff
@@ -261,13 +267,15 @@ impl MqttState {
     }
 
     fn handle_incoming_pubrel(&mut self, pubrel: &PubRel) -> Result<(), StateError> {
-        match mem::replace(&mut self.incoming_pub[pubrel.pkid as usize], None) {
-            Some(_) => {
+        match self.incoming_pub.get_mut(pubrel.pkid as usize) {
+            Some(pkid) => {
+                *pkid = None;
+
```
Comment (de-sh) [inline rumqttc/src/state.rs:273]:
```suggestion
        let publish = self
            .outgoing_pub
            .get_mut(pubrec.pkid as usize)
            .ok_or(StateError::Unsolicited(pubrec.pkid))?;
        match publish.take() {
            Some(_) => {
```

**Review by de-sh — APPROVED** [review]
LGTM

**Pull request merged**